### PR TITLE
React set up and basic component rendered

### PR DESCRIPTION
**Pull request by sharkhouse23**
@tarikbellamine can you review my PR?

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -0,0 +1,74 @@
+import React from 'react';
+import $ from 'jquery';
+import SimilarCourses from './SimilarCourses.jsx';
```
Comment (tarikbellamine) [inline client/src/components/App.jsx]: Consider putting this file in a components folder

File: client/src/components/App.jsx [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+  constructor(props) {
+    super(props);
+    this.state = {
+      courses: [],
+    }
+
+    this.retrieveFromDB = this.retrieveFromDB.bind(this);
+  }
+
+  // retrieveFromDB(event) {
```
Comment (tarikbellamine) [inline client/src/components/App.jsx]: delete comments

File: database/index.js [hunk 3]
```diff
@@ -1,71 +1,129 @@
...
+}
 
-	connection.query('Select * from Courses', function (error, results) {
-		if (error) {
-			callback(error, null);
-		} else {
-			callback(null, results)
-		}
-	})
+const arrayOfPurchasesForStudents = function(courseId, callback) {
```
Comment (tarikbellamine) [inline database/index.js]: Why do you need to pass a callback as a param in your chain of promises? Shouldn't you be able to call the function as part of the chain?

File: server/server.js [hunk 4]
```diff
@@ -14,28 +14,29 @@ app.use(express.static('public'));
 
 
 // GET method route
-// app.get('/', function (req, res) {
-//   db.getAll((err, results) => {
+app.get('/input', function (req, res) {
+  db.arrayOfPurchasesForStudents(1, (err, results) => {
```
Comment (tarikbellamine) [inline server/server.js]: What is the significance of 1?

File: database/index.js [hunk 5]
```diff
@@ -1,71 +1,129 @@
...
+      .catch(err => { 
+        callback(err);
+      })
+  })
+  .catch(error => {
+    callback(error);
+  })
 }
 
+arrayOfPurchasesForStudents();
```
Comment (tarikbellamine) [inline database/index.js]: Why are you invoking the function here?

**Comment by tarikbellamine** [comment]
@sharkhouse23 made a few comments. From a stylistic perspective, I recommend deleting out comments and console logs once you've gotten everything functioning. Makes the code easier to follow.

From a functionality perspective, it was a little hard to follow your db querying because of the mix of error first callbacks and promises. I wouldn't suggest refactoring at this point if the queries work but might be a good idea to use only one method next time.

File: client/src/components/App.jsx (hunk 1 above)
Comment (sharkhouse23) [reply client/src/components/App.jsx]: aren't all the jsx files in the components folder?

File: client/src/components/App.jsx (hunk 2 above)
Comment (sharkhouse23) [reply client/src/components/App.jsx]: good point, will do that right now

File: database/index.js (hunk 5 above)
Comment (sharkhouse23) [reply database/index.js]: good point, this is probably not the best place to invoke this func

File: database/index.js (hunk 3 above)
Comment (sharkhouse23) [reply database/index.js]: depending on how you set up your db query and server requests, you may not need callback as a parameter in the arrayOfPurchasesForStudents func

File: server/server.js (hunk 4 above)
Comment (sharkhouse23) [reply server/server.js]: no significance, it's purely for console.logging purposes. Eventually I'll have to make the functions dynamic enough to handle any courseId provided

**Comment by sharkhouse23** [comment]
@tarikbellamine Thanks for your feedback. Will make the suggested changes accordingly.
